dams-local-client/src/api.rs

```diff
@@ -251,58 +285,77 @@ impl Session {
     }
 
     async fn do_register<T: CryptoRng + RngCore>(
+        client: &mut DamsRpcClient<Channel>,
         rng: &mut T,
         user_id: &UserId,
         password: &Password,
         server: &KeyMgmtAddress,
         config: &self::Config,
-    ) -> Result<(), anyhow::Error> {
-        // Connect with the server...
-        let (_session_key, chan) = connect(config, server)
-            .await
-            .map_err(|_| SessionError::ServerConnectionFailed)?;
+    ) -> Result<Response<ServerRegisterFinish>, Status> {
+        // Create channel to send messages to server after connection is established via
+        // RPC
+        let (tx, rx) = mpsc::channel(2);
+        let stream = ReceiverStream::new(rx);
 
-        // ...and select the Register session
-        let chan = chan
-            .choose::<1>()
-            .await
-            .map_err(|_| SessionError::SelectRegisterFailed)?;
+        // Server returns its own channel that is uses to send responses
+        let mut server_response = client.register(stream).await?.into_inner();
 
         let client_registration_start_result =
             ClientRegistration::<OpaqueCipherSuite>::start(rng, password.as_bytes())
-                .map_err(|_| SessionError::RegisterStartFailed)?;
-
-        let chan = chan
-            .send(RegisterStart::new(
-                client_registration_start_result.message,
-                user_id.clone(),
-            ))
+                .map_err(|_| Status::aborted("RegistrationStart failed"))?;
+
+        // Send start message to server
+        let client_registration_start_message =
+            dams::serialize_to_bytes(&client_registration_start_result.message)?;
+        let register_start =
+            Self::client_register_start(client_registration_start_message, user_id);
+        tx.send(register_start)
             .await
-            .map_err(|_| SessionError::RegisterStartFailed)?;
-
-        offer_abort!(in chan as Client);
-
-        let (registration_response, chan) = chan
-            .recv()
-            .await
-            .map_err(|_| SessionError::RegisterStartFailed)?;
-
+            .expect("Handle weird error type");
```

Comment:
```suggestion
            .map_err(|e| Status::aborted(e.to_string()))?;
```

`expect` will panic if there's an error here.